pr_agent/git_providers/bitbucket_provider.py

```diff
@@ -263,3 +286,89 @@ def publish_labels(self, pr_types: list):
     # bitbucket does not support labels
     def get_labels(self):
         pass
+
+    def get_issue(self, issue_url):
+        workspace_slug, repo_name, original_issue_number = self._parse_issue_url(issue_url)
+        issue = self.bitbucket_client.repositories.get(workspace_slug, repo_name).issues.get(original_issue_number)
+        return issue, original_issue_number
+    
+    def get_issue_url(self, issue):
+        return issue._BitbucketBase__data['links']['html']['href']
+    
+    def get_issue_body(self, issue):
+        return issue.content['raw']
+    
+    def get_issue_number(self, issue):
+        return issue.id
+    
+    def get_issue_comment_body(self, comment):
+        return comment['content']['raw']
+    
+    def get_issue_comment_user(self, comment):
+        return comment['user']['display_name']
+    
+    def get_issue_created_at(self, issue):
+        return str(issue.created_on)
+    
+    def get_username(self, issue, issue_url):
+        workspace_slug, repo_name, original_issue_numbers = self._parse_issue_url(issue_url)
+        return workspace_slug
+
+    
+    def get_repo_issues(self, repo_obj):
+        return repo_obj._Repository__issues.each()
+
+   
+    def parse_issue_url_and_get_comments(self, issue):
+        workspace_slug, repo_name, original_issue_number = self._parse_issue_url(issue)
+        import requests
+
+        url = f"https://api.bitbucket.org/2.0/repositories/{workspace_slug}/{repo_name}/issues/{original_issue_number}/comments"
+
+        payload = {}
+        headers = {}
+
+        response = requests.request("GET", url, headers=headers, data=payload)
```

Comment:
**Suggestion:** Consider using a helper function to handle the HTTP requests, as this operation is repeated multiple times in the code. This will improve code readability and maintainability. [maintainability]
```suggestion
        def make_request(self, method, url, headers={}, payload={}):
            response = requests.request(method, url, headers=headers, data=payload)
            return response.json()['values']

        url = f"https://api.bitbucket.org/2.0/repositories/{workspace_slug}/{repo_name}/issues/{original_issue_number}/comments"
        return self.make_request("GET", url)
```